lib/bundler/resolver.rb

```diff
@@ -296,14 +295,23 @@ def sort_dependencies(dependencies, activated, conflicts)
         name = name_for(dependency)
         [
           activated.vertex_named(name).payload ? 0 : 1,
-          @prereleases_cache[dependency.requirement] ? 0 : 1,
-          dependency.requirements_list.size,
+          amount_constrained(dependency),
           conflicts[name] ? 0 : 1,
           activated.vertex_named(name).payload ? 0 : search_for(dependency).count,
         ]
       end
     end
 
+    def amount_constrained(dependency)
+      @amount_constrained ||= {}
+      @amount_constrained[dependency.name] ||= begin
+        base_dep = Dependency.new dependency.name, '>= 0.a'
+        all = search_for(DepProxy.new base_dep, dependency.__platform)
```

Comment:
Doesn't calling __platform cause a request to download the gemspec? I worry that this is really, really slow when we don't already have the platform of the dep...